python/pyarrow/_flight.pyx

```diff
@@ -1150,6 +1156,38 @@ cdef class FlightClient(_Weakrefable):
                 self.client.get().Authenticate(deref(c_options),
                                                move(handler)))
 
+    def authenticateBasicToken(self, username, password,
```

Comment:
Do you need to add a description for self in the list of parameters?

Reply:
self is never documented - it's implicitly passed by Python (it's equivalent to `this` in Java et al)